Decisions.TruBot/Data/TruBotRecordedBot.cs

```diff
@@ -0,0 +1,81 @@
+using DecisionsFramework.Data.ORMapper;
+using DecisionsFramework.Design.Properties;
+using DecisionsFramework.ServiceLayer;
+using DecisionsFramework.Utilities;
+
+namespace Decisions.TruBot.Data
+{
+    [ORMEntity("trubot_recorded_bots")]
+    public class TruBotRecordedBot : AbstractFolderEntity
+    {
+        public TruBotRecordedBot() {}
+
+        [ORMPrimaryKeyField]
+        [PropertyHidden]
+        public string Id { get; set; }
+        
+        [ORMField]
+        public int BotId { get; set; }
+        
+        [ORMField]
+        public string BotName { get; set; }
+        
+        [ORMField]
+        public DateTime InitializedOn { get; set; }
+        
+        [ORMField]
+        public DateTime LastRunOn { get; set; }
+        
+
+        public override void BeforeSave()
+        {
+            if (string.IsNullOrEmpty(Id))
+            {
+                Id = IDUtility.GetNewIdString();
+            }
+            base.BeforeSave();
+        }
+
+        static ORM<TruBotRecordedBot> orm = new();
+
+        public TruBotRecordedBot(int botId)
+        {
+            TruBotRecordedBot bot = GetTruBotRecordByBotId(botId);
+
+            if (bot != null)
+            {
+                orm.Fetch(typeof(TruBotRecordedBot));
+
+                bot = GetTruBotRecordByBotId(botId);
```

Comment:
When you create a RecordedBot, you're looking for another RecordedBot with the same botId?
This seems strange and circular. Lets talk more about what the intention is here and how this should work, I don't fully understand the pattern.

Reply:
1. Create a new bot record if one doesn't exist
2. If it does exist, then update the existing record instead of creating a new one

The intention is to keep a record of existing (previously used) bots since we cannot fetch a list of all active bots with the TruBot API.